app.js

```diff
@@ -1,18 +1,75 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
 
 // require spotify-web-api-node package here:
-
+const SpotifyWebApi = require("spotify-web-api-node");
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
 
 // Our routes go here:
+app.get("/", (req, res) => {
+  res.render("index", { title: "index" });
+});
+
+app.get("/artist-search", (req, res) => {
+  spotifyApi
+    .searchArtists(req.query.search)
+    .then((data) => {
+      // ----> 'HERE WHAT WE WANT TO DO AFTER RECEIVING THE DATA FROM THE API'
+      res.render("artist-search-results", {
+        artists: data.body.artists.items,
+        query: req.query.search,
+      });
+    })
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+});
+
+app.get("/albums/:artistId", (req, res, next) => {
+  spotifyApi
+    // This was were I thought there was something wrong with the Api, but after we spoke I found there were two
+    // getArtistAlbums" in the documentation and the other one works fine
+    .getArtistAlbums(req.params.artistId)
+    .then(
+      function (data) {
+        res.render("albums", { albums: data.body.items });
+      },
+      function (err) {
+        console.error(err);
+      }
+    );
+});
+
+app.get("/tracks/:albumId", (req, res, next) => {  
+  spotifyApi.getAlbumTracks(req.params.albumId).then(
+    function (data) {
```

Comment:
I personall dislike function syntax like this within .then blocks. => arrows are so much shorter and to me at least, more beautiful. 

also, maybe try writing this code with async/await as a practice.

Reply:
Yeah, I think I agree. If I remember correctly this is how it was done in the material on IronHack, so I just wanted to try it out once, too :)